### PR TITLE
*: Cut v1.6.0-rc.0

### PR DESCRIPTION
**Pull request by mxinden**
**Which issue(s) this PR fixes**:
Relates to https://github.com/kubernetes/kube-state-metrics/issues/697

### CODE REVIEW

**Comment by mxinden** [comment]
Still in _work-in-progress_ mode, as I have not yet done any benchmarking.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+## v1.6.0 / 2019-03-18
+
+* [FEATURE] Add `kube-replicaset_labels` to replicaset collector (#638)
+* [FEATURE] Add ingresses collector (#640)
+* [FEATURE] Add `kube_persistentvolumeclaim_access_mode` metric (#673)
+* [FEATURE] Add `kube_persistentvolume_capacity` metric (#674)
+* [FEATURE] Add `kube_job_owner` metric (#681)
+
```
Comment (tariq1890) [inline CHANGELOG.md]:
```suggestion
* [FEATURE] Add collector for csr (#650)
```

**Comment by tariq1890** [comment]
Hey @mxinden , Any updates on this :) ?

**Comment by mxinden** [comment]
Sorry for the delay @tariq1890, I have been sick for a week.

I don't seem to have permissions to create branches. @brancz is on vacation. @andyxning are you able to do so? In that case I would do some last changes to this pull request and let you proceed if you like.

**Comment by tariq1890** [comment]
No problem at all. Hope you get well soon @mxinden :)

**Comment by tariq1890** [comment]
Hi @andyxning Are we able to create the upstream `release-1.6` branch for this PR?

**Comment by mxinden** [comment]
I reached out to our internal scalability team for some performance tests. I will keep you all  updated.

**Comment by brancz** [comment]
Typically we cut a release candidate and then performed the performance tests on that. There's also a scalability team within kubernetes we can reach out to.

File: CHANGELOG.md (hunk 1 above)
Comment (mxinden) [reply CHANGELOG.md]: Added, thanks.

**Comment by mxinden** [comment]
@brancz release candidate sounds good as well. I adjusted the pull request. I don't have permissions to proceed in the release process. Would you mind @brancz?

**Comment by mxinden** [comment]
Once we decided on https://github.com/kubernetes/kube-state-metrics/pull/701 we can continue here.

**Comment by tariq1890** [comment]
@mxinden Now that #701 is merged. We'll now need to update the K8s version support matrix in the docs.

**Comment by mxinden** [comment]
I have compared kube-state-metrics *v1.4.0*, *v1.5.0* and *v1.6.0-rc.0* (including https://github.com/kubernetes/kube-state-metrics/pull/701) on a 9 node cluster with AWS *m5.large* compute instances. During the comparison a test script would create and delete pods ranging from 250 - 5000 pods total. In addition the cluster had 8000 secrets constantly. Kube-state-metrics instances are scraped every 2s.

### Scrape duration

Scrape duration seems to differ little between *v1.5.0* and *v1.6.0-rc.0*, but the random but repeated spikes attract attention. This could have many reasons, among others external once, e.g. noisy neighbours. I will leave the cluster running for an additional night.

image


### CPU usage

CPU usage seems to be the same between *v1.5.0* and *v1.6.0-rc.0*.

image

### Memory usage

Memory usage seems to be the same between *v1.5.0* and *v1.6.0-rc.0*.

image

### Conclusion

Overall *v1.6.0-rc.0* looks good from a performance perspective. I will look further into the random scrape duration spikes. As this is a release candidate, I think this is good to go from a performance perspective.

While CPU profiling the new release candidate under load I found two optimizations that we could apply. I will follow up  with an additional pull request. Given that *v1.6.0-rc.0* doesn't seem to introduce any performance regressions, the upcoming patch does not need to be included in the release.

**Comment by brancz** [comment]
/lgtm

Just wondering why CI is already trying to compare `release-1.6` with the latest release?

**Comment by mxinden** [comment]
> @mxinden Now that #701 is merged. We'll now need to update the K8s version support matrix in the docs.

Adding hold.

/hold

@tariq1890 do you have a chance to look into that? Otherwise I will do it tomorrow.

**Comment by mxinden** [comment]
> Just wondering why CI is already trying to compare release-1.6 with the latest release?

@brancz the benchmark script reads the `VERSION` file, and tries to find the matching release file. This works for all pull requests except minor and major release pull requests, given that the branch does not yet exist and given that it would only compare to itself.

Detailed error message:

```
### Testing release-1.6
error: pathspec 'release-1.6' did not match any file(s) known to git.
make: *** [test-benchmark-compare] Error 1
The command "make test-benchmark-compare" exited with 2.
```

I can't come up with a better solution right now, as doing version parsing on the `VERSION` file to get the previous one for major and minor releases seems a bit too much magic to me. I would suggest delaying it and ignoring the failure here for now. What do you all think?

---

This is waiting for https://github.com/kubernetes/kube-state-metrics/pull/711 and possibly https://github.com/kubernetes/kube-state-metrics/pull/709.

**Comment by brancz** [comment]
how about we already create the `release-1.6` branch off of current master?

**Comment by mxinden** [comment]
:+1: Tests are green now. Any follow ups or is this good to go for all of you?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
+## v1.6.0-rc.0 / 2019-04-01
+
+* [FEATURE] Add `kube-replicaset_labels` to replicaset collector (#638)
```
Comment (brancz) [inline CHANGELOG.md]: typo `kube-replicaset_labels` -> `kube_replicaset_labels`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
+* [FEATURE] Add ingresses collector (#640)
```
Comment (brancz) [inline CHANGELOG.md]: We typically end lines with a period

File: CHANGELOG.md (hunk 2 above)
Comment (mxinden) [reply CHANGELOG.md]: Fixed.

File: CHANGELOG.md (hunk 3 above)
Comment (mxinden) [reply CHANGELOG.md]: Oh, missed that, thanks!

**Comment by brancz** [comment]
Sorry for dropping the ball on this. Let's continue this on Monday.

/hold
/lgtm

**Comment by brancz** [comment]
Shall we continue with this? (@mxinden if you agree feel free to remove the hold)

**Pull request merged**